test/tests/functional/pbs_test_tpp.py

```diff
@@ -386,26 +389,21 @@ def test_multiple_comm_with_mom(self):
         self.momA = self.moms.values()[0]
         self.momB = self.moms.values()[1]
         self.comm1 = self.comms.values()[0]
+        self.comm2 = self.comms.values()[1]
```

Comment:
There seems to be mismatch in requirements decorator data, test nodes configuration, and the number of nodes needed here.
1. By default no_mom_on_server=False i.e. there is a mom on server host: this calls for docstring correction
2. That said, this test would only need 3 hosts: hence no need of self.hostD
3. Also PBS_LEAF_ROUTERS is needed only at the mom-only host and not on both moms, since one mom each is reported by each comm